src/server/pps/cmds/cmds.go

```diff
@@ -846,19 +852,27 @@ func (arr ByCreationTime) Less(i, j int) bool {
 	return false
 }
 
-// pushImage pushes an image as registry/user/image. Registry and user can be
-// left empty.
-func pushImage(registry string, username string, password string, image string) (string, error) {
+type dockerManager struct {
+	client *docker.Client
+	config docker.AuthConfiguration
+	registry string
+	repo string
+	sourceTag string
+	destTag string
+}
```

Comment:
I've removed the class in favor of just functions. It does look cleaner, albiet a little wordier since all the necessary parameters are passed around. Let me know what you think.